aquadoggo/src/materializer/tasks/dependency.rs

```diff
@@ -1,9 +1,293 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
+use p2panda_rs::document::DocumentViewId;
+
 use crate::context::Context;
-use crate::materializer::worker::TaskResult;
+use crate::db::traits::DocumentStore;
+use crate::materializer::worker::{Task, TaskError, TaskResult};
 use crate::materializer::TaskInput;
 
-pub async fn dependency_task(_context: Context, _input: TaskInput) -> TaskResult<TaskInput> {
-    Ok(None)
+/// Helper method for retrieving a document view by it's document view id and if it doesn't exist in
+/// the store, composing a "reduce" task for this specific document view.
+async fn construct_relation_task(
```

Comment:
You could move this function further down the file as it's a helper and instead put `dependency_task` at the top as that is what this file is about.